test/unittest/pointertest.cpp

```diff
@@ -503,7 +503,7 @@ TEST(Pointer, Assignment) {
         EXPECT_STREQ("0", q.GetTokens()[1].name);
         EXPECT_EQ(0u, q.GetTokens()[1].index);
         EXPECT_NE(&p.GetAllocator(), &q.GetAllocator());
-        q = q;
+        q = static_cast<Pointer>(q);
```

Comment:
According to the Clang documentation, the cast should be to a **reference** in case self-assignment is wanted, which is the case here:
```suggestion
        q = static_cast<const Pointer &>(q);
```